### PR TITLE
add schedule action

### PR DESCRIPTION
**Pull request by seaify**
https://chowbus.atlassian.net/browse/BAC-195

### CODE REVIEW

File: app/models/delay_henka/scheduled_action.rb [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        arguments: arguments,
+        submitted_by_id: by_id,
+        schedule_at: schedule_at
+      )
+    end
+
+    def apply_action
+      if actionable
+        update(state: STATES[:COMPLETED])
+        actionable.send(method_name, *arguments)
```
Comment (seaify) [inline app/models/delay_henka/scheduled_action.rb]: @zinosama i thought during exec the method, exception might happen, so need set state in advance

**Comment by seaify** [comment]
@zinosama i don't think there's need to check if the method_name exists and arguments legal, how do you think?

File: app/models/delay_henka/scheduled_action.rb [hunk 2]
```diff
@@ -0,0 +1,44 @@
+module DelayHenka
+  class ScheduledAction < ApplicationRecord
+
+    STATES = {
+      STAGED: 'staged',
+      COMPLETED: 'completed',
+    }
```
Comment (zinosama) [inline app/models/delay_henka/scheduled_action.rb]: missing ERROR

File: db/migrate/20190408130947_create_delay_henka_scheduled_actions.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
+class CreateDelayHenkaScheduledActions < ActiveRecord::Migration[5.2]
+  def change
+    create_table :delay_henka_scheduled_actions do |t|
+      t.references :actionable, polymorphic: true, null: false, index: { name: 'actionable_index' }
+      t.string :method_name
```
Comment (zinosama) [inline db/migrate/20190408130947_create_delay_henka_scheduled_actions.rb]: missing not null

File: db/migrate/20190408130947_create_delay_henka_scheduled_actions.rb [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+      t.text :arguments, array: true, default: []
+      t.string :state, null: false, index: true
+      t.string :error_message
+      t.integer :submitted_by_id, null: false
+      t.datetime :schedule_at, index: true
```
Comment (zinosama) [inline db/migrate/20190408130947_create_delay_henka_scheduled_actions.rb]: missing not null

File: app/models/delay_henka/scheduled_action.rb [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+    belongs_to :actionable, polymorphic: true
+
+    validates :submitted_by_id, presence: true
+    validates :schedule_at, presence: true
```
Comment (zinosama) [inline app/models/delay_henka/scheduled_action.rb]: validates :submitted_by_id, : schedule_at, presence: true

**Comment by zinosama** [comment]
alright i'll work on this one. don't worry about the changes @seaify

**Pull request merged**